lib/navigation/preprocess_imu.cpp

```diff
@@ -1,56 +1,120 @@
 #include "preprocess_imu.hpp"
 
-namespace hyped::navigation {
+#include <cmath>
```

Comment:
You're missing headers. You'll need <algorithm>, <iterator> and if it's not already in the header file, <array> too.